rootfs/usr/bin/hwctl

```diff
@@ -17,11 +17,17 @@ function system-info {
 
 	# CPU
 	cat /proc/cpuinfo > /tmp/cpuinfo.txt
-	CPU_VENDOR=$(grep 'vendor_id' /tmp/cpuinfo.txt | head -n1 | cut -d':' -f2 | sed 's/^ //')
-	CPU_NAME=$(grep 'model name' /tmp/cpuinfo.txt | head -n1 | cut -d':' -f2 | sed 's/^ //')
-	CPU_PHYSICAL=$(grep 'cpu cores' /tmp/cpuinfo.txt | head -n1 | cut -d':' -f2 | tr -d ' ')
-	CPU_SPEED="$(lscpu | grep 'CPU max MHz' | head -n 1 | cut -d':' -f2 | tr -d ' ' | cut -d'.' -f1)"
-	CPU_ARCH="$(lscpu | grep 'Architecture:' | head -n 1 | cut -d':' -f2 | tr -d ' ')"
+	lscpu > /tmp/lscpu.txt
+	CPU_ARCH="$(grep 'Architecture:' /tmp/lscpu.txt | head -n 1 | cut -d':' -f2 | tr -d ' ')"
+	if [[ "${CPU_ARCH}" == "x86_64" ]]; then
+		CPU_VENDOR=$(grep 'vendor_id' /tmp/cpuinfo.txt | head -n1 | cut -d':' -f2 | sed 's/^ //')
+		CPU_NAME=$(grep 'model name' /tmp/cpuinfo.txt | head -n1 | cut -d':' -f2 | sed 's/^ //')
+	elif [[ "${CPU_ARCH}" == "aarch64" ]]; then
+		CPU_VENDOR=$(grep 'Vendor ID' /tmp/lscpu.txt | head -n1 | cut -d':' -f2 | sed 's/[[:space:]]//g')
+		CPU_NAME=$(grep 'Model name' /tmp/lscpu.txt | head -n1 | cut -d':' -f2 | sed 's/[[:space:]]//g')
+	fi
+	CPU_PHYSICAL=$(grep -c ^processor /tmp/cpuinfo.txt)
```

Comment:
This appears to give the number of threads, not physical cores.

Reply:
A simple alternative that will work on all architectures:

```bash
ls /sys/bus/cpu/devices | wc -l
```